Add seeders/leechers to notifiers and cache. Prepare SR for "manual snatch" feature

fernandog opened the pull request:
- Adds seeder/leechers in notifiers message
- Adds seeders/leechers in the "Downloading..." INFO log message - more visible to users
- Prepare cache.db to show seeders/leechers when feature **"manual snatch"** is implemented so users can choose releases based on seeds/leechers
- Update cache.db when parsing items at DailySearcher so users have updated seeders/leechers when manual snatch

**Next step** for another PR: create a pop up showing cache.db results for that show so user can manually snatch a result

http://feathub.com/SickRage/SickRage/+23

SickRage2 commented:
Thanks for the pull request! Before a real human comes by, please make sure your PR has all of the below criteria checked

- [ ] Give a description on what the PR is for.
- [x] Make sure your PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review

Please make sure you also read [contribution guide](https://github.com/SickRage/SickRage/blob/master/contributing.md) and followed all of the steps.

Thanks!

miigotu commented:
No.

fernandog commented:
which number?

miigotu commented:
Cool so lets take something I said no about and mix some other features in and make it worse...

fernandog commented:
@miigotu
Are you sure they are not related? Think again:

manual snatch feature need cached seeders/leechers to avoid ddos providers. or are we going to hit providers for every manual snatch?

And remember that feature request is the #1 FR. So what is the real reason for closing this as I did all changes as @labrys said?

So instead of saying 'no', what is you solution to make it work and/or add that FR? Or why not 'do not merge' until more improvents to code?

I bet you and @SickRage/collaborators that wont find issues in my code. Say your bet.

miigotu commented:
I can point out at least 5 bugs in this PR, besides the fact that you are trying to force your seeders/leechers in notifications.

You need to learn to accept criticism, how to take no for an answer and move on, and how to manage risk.

idgaf if you PR it 10 times, the answer will be the same when you just keep PR the same shit.

fernandog commented:
I fixed the issues pointed in the other PR
I not seeing you pointing issues in the code.
Just empty criticism

As you already checked where are those 5 bugs? Improvements dont count.

miigotu commented on sickrage/providers/torrent/TorrentProvider.py:
```diff
@@ -96,6 +97,20 @@ def _get_size(self, item):
 
+    def _get_result_info(self, item):
+        if isinstance(item, (dict, FeedParserDict)):
+            seeders = item.get('seeders', '')
```
no bug?

fernandog replied:
your original code in get_title

miigotu commented on sickrage/providers/torrent/TorrentProvider.py:
```diff
@@ -96,6 +97,20 @@ def _get_size(self, item):
+    def _get_result_info(self, item):
+        if isinstance(item, (dict, FeedParserDict)):
+            seeders = item.get('seeders', '')
+            leechers = item.get('leechers', '')
```
or here?

fernandog replied:
your original code in get_title

miigotu commented on sickrage/providers/torrent/TorrentProvider.py:
```diff
@@ -96,6 +97,20 @@ def _get_size(self, item):
+            seeders = item[3]
+            leechers = item[4]
+        else:
+            seeders = ''
```
or here?

fernandog replied:
already explained

miigotu commented on sickrage/providers/torrent/TorrentProvider.py:
```diff
@@ -96,6 +97,20 @@ def _get_size(self, item):
+            leechers = item[4]
+        else:
+            seeders = ''
+            leechers = ''
```
what about here?

fernandog replied:
its your original code here! lol
made as @labrys said. make a specific function for it

miigotu commented on sickrage/providers/GenericProvider.py:
```diff
@@ -443,6 +446,9 @@ def _get_size(self, item):  # pylint: disable=unused-argument,no-self-use
         return ''
 
+    def _get_result_info(self, item):  # pylint: disable=no-self-use
+        return '', ''
```
or this?

fernandog replied:
what is wrong?

NZB don't have extra info. what should be returned?

miigotu replied:
methods ALWAYS need to return the same type, regardless of what is subclassing it.

You could have saved yourself a whole lot of time and effort if you would have just accepted the no to begin with.  You could have saved me a whole lot of time and effort also, and let me waste that time and effort on more effective shit.

fernandog replied:
not something to close PR isn't?

miigotu commented on sickbeard/tvcache.py:
```diff
@@ -60,6 +60,14 @@ def __init__(self, providerName):
 
+            # add seeders column to table if missing
+            if not self.hasColumn(providerName, 'seeders'):
+                self.addColumn(providerName, 'seeders', "NUMERIC", "")
```
here?

fernandog replied:
easy fix. not something to close PR isn't?

miigotu commented on sickbeard/tvcache.py:
```diff
@@ -60,6 +60,14 @@ def __init__(self, providerName):
+
+            # add leechers column to table if missing
+            if not self.hasColumn(providerName, 'leechers'):
+                self.addColumn(providerName, 'leechers', "NUMERIC", "")
```
here?

fernandog replied:
as I said: empty = nzb.  0 seeds is zero seeds. can't be INT

miigotu commented on sickbeard/search_queue.py:
```diff
@@ -274,7 +283,10 @@ def run(self):
                 for result in searchResult:
                     # just use the first result for now
-                    logger.log(u"Downloading " + result.name + " from " + result.provider.name)
+                    if result.seeders and result.leechers:
```
so when leechers is 0 dont show seeder count?

fernandog replied:
easy fix. not something to close PR isn't?

miigotu commented on sickbeard/search_queue.py:
```diff
@@ -224,7 +230,10 @@ def run(self):
                     for result in searchResult:
                         # just use the first result for now
-                        logger.log(u"Downloading " + result.name + " from " + result.provider.name)
+                        if result.seeders and result.leechers:
```
also here?

fernandog replied:
multiple point the same code doesn't count as bug.
if seeders = "" means it's NZB. 0 is for zero seeds

miigotu commented on sickbeard/search_queue.py:
```diff
@@ -180,7 +183,10 @@ def run(self):
             if searchResult:
                 # just use the first result for now
-                logger.log(u"Downloading " + searchResult[0].name + " from " + searchResult[0].provider.name)
+                if searchResult[0].seeders and searchResult[0].leechers:
```
and here?

fernandog replied:
where? what's wrong?

miigotu commented on sickbeard/search_queue.py:
```diff
@@ -142,7 +142,10 @@ def run(self):
                 for result in foundResults:
                     # just use the first result for now
-                    logger.log(u"Downloading " + result.name + " from " + result.provider.name)
+                    if result.seeders and result.leechers:
```
and here?

fernandog replied:
where? what's wrong?

miigotu commented on sickbeard/search.py:
```diff
@@ -169,11 +169,14 @@ def snatchEpisode(result, endStatus=SNATCHED):
             try:
-                notifiers.notify_snatch(curEpObj._format_pattern('%SN - %Sx%0E - %EN - %QN') + " from " + result.provider.name)
-            except:
+                if result.seeders and result.leechers:
```
and here?

fernandog replied:
where?! what wrong?

fernandog replied:
easy fix. not something to close PR isn't?

miigotu commented on sickbeard/classes.py:
```diff
@@ -103,6 +103,12 @@ def __init__(self, episodes):
         self.size = -1
 
+        # seeders of the release
+        self.seeders = ""
```
so seeders and leechers is a string now?

fernandog replied:
can't be int. cause 0 means not seeds.
seeds = "" - nzb

miigotu commented on sickbeard/classes.py:
```diff
@@ -103,6 +103,12 @@ def __init__(self, episodes):
+        self.seeders = ""
+
+        # leechers of the release
+        self.leechers = ""
```
?

fernandog replied:
what didnt you understand? I will explain

miigotu commented:
Talk some more shit?

miigotu commented:
I dont give a shit if you fix every bit of what I point out. We arent making these ridiculous changes just to show seeds/leechers in a NOTIFICATION.

fernandog commented:
you didn't read the PR

this is for notification **AND feature manual snatch feature**

miigotu commented:
So it is ok to mix 2 PR into one to try and sneak in something I SAID NO ABOUT? Even if the answer was yes mixing features is not ok. just stfu and stop spamming me now.

I swear that you could argue that the sun wont come up, and then the sun comes up and you will argue that it wont go down. Both ways you are wrong.

miigotu commented:
Every single one of those is a bug, not an improvement. The code is incorrect. The entire PR is not necessary at all to begin with.

Besides the fact that those arent the only problems here, those are just the first 15 or so that I found in 30 seconds.

adaur commented:
I don't know if the code is good, but the idea is. We'll need to have the seeders and leechers in the cache to display them when one needs to choose its release to download.

fernandog commented:
@adaur 
I removed the seeders/leechers from notifiers in my local branch and fixed all points mii said.

https://github.com/fernandog/SickRage/commit/abce8dc0febec9bfb5679e62f0a4a1e69bfba651

duramato commented:
@fernandog size also needs to be cached like seeder/leechers , right @adaur ?

fernandog commented:
Added new commit to it:
https://github.com/fernandog/SickRage/commits/seeds_in_notify
